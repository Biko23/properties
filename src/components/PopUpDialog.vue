<template>
<v-dialog transition="dialog-top-transition" persistent v-model="messageDialog" max-width="600">
    <template>
        <v-card>
            <v-toolbar :color="color" dark>{{title}}</v-toolbar>
            <v-card-text>
                {{message}}
            </v-card-text>
            <v-card-actions class="justify-end">
                <!-- <v-btn text @click="closeDialog">ok</v-btn> -->
                <slot></slot>
            </v-card-actions>
        </v-card>
    </template>
</v-dialog>
</template>

<script>
export default {
    name: "PopComponent",
    props: {
       messageDialog: {
           type: Boolean,
           default: false
       },
        title: {
           type: String,
           default: "No Title"
       },
       message: {
           type: String,
           default: "No message available"
       },
       color: {
           type: String,
           default: "primary"
       }
    }
}
</script>
