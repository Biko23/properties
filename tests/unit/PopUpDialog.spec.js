import { createLocalVue, shallowMount } from '@vue/test-utils'
import Vuetify from 'vuetify'
import PopUpDialog from '@/components/PopUpDialog.vue'

describe('PopUpDialog.vue', () => {
    const localVue = createLocalVue()
    localVue.use(Vuetify)
    let vuetify

    beforeEach(() => {
        vuetify = new Vuetify()
    })

    const mountFunction = options => {
        return shallowMount(PopUpDialog, {
            localVue,
            vuetify,
            ...options
        })
    }

    it('should have a match snapshot', () => {
        const wrapper = mountFunction()
        expect(wrapper.html()).toMatchSnapshot()
    })

    it('Should contains default props values', () => {
        const wrapper = mountFunction()

        expect(wrapper.find('[data-testid="title-element"]').text()).toBe('No Title')
        expect(wrapper.find('[data-testid="message-element"]').text()).toBe('No message available')
        expect(wrapper.find('[data-testid="title-element"]').props().color).toBe('primary')
    })

    it('Should contains modified props values', () => {
        const wrapper = mountFunction({
            propsData: {
                title: 'New Title',
                message: 'New Message',
                color: 'Green'
            }
        })

        expect(wrapper.find('[data-testid="title-element"]').text()).toBe('New Title')
        expect(wrapper.find('[data-testid="message-element"]').text()).toBe('New Message')
        expect(wrapper.find('[data-testid="title-element"]').props().color).toBe('Green')
    })
})

