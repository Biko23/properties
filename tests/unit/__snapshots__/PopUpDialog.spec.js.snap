// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopUpDialog.vue should have a match snapshot 1`] = `
<v-dialog-stub opendelay="0" closedelay="0" contentclass="" maxwidth="600" origin="center center" persistent="true" retainfocus="true" transition="dialog-top-transition" width="auto">
  <v-card-stub loaderheight="4" tag="div">
    <v-toolbar-stub color="primary" dark="true" tag="header" extensionheight="48" src="" data-testid="title-element">No Title</v-toolbar-stub>
    <v-card-text-stub datatestid="message-element" data-testid="message-element">
      No message available
    </v-card-text-stub>
    <v-card-actions-stub class="justify-end"></v-card-actions-stub>
  </v-card-stub>
</v-dialog-stub>
`;
